drivers/public_reports/app/models/public_reports/state_level_homelessness.rb

```diff
@@ -238,13 +238,13 @@ def parsed_pre_calculated_data
     private def end_iteration(date)
       return date.end_of_quarter unless yearly?
 
-      return date.end_of_year
+      return [date.end_of_year, filter_object.end_date].min
```

Comment:
I feel like 
```ruby
def end_date(date)
  [date.end_of_year, filter_object.end_date].min
end
```
Should get pulled out to a method to ensure consistency? Also, wondering why RuboCop didn't complain about the `return`?

```suggestion
      end_date(date)

Reply:
Good question about rubocop, that seems odd.  I thought of doing that, but it's not identical all the way through the file.  There are a bunch of calls to `end_date` as a variable for different end dates, so for now I left it.  It could maybe be `year_end_date` or something more specific.  Let's refactor after we confirm it fixes the issue.